library/spdm_requester_lib/libspdm_req_get_measurement_extension_log.c

```diff
@@ -0,0 +1,426 @@
+ * @param  measurement_record         A pointer to HEM measurement block.
+ * @param  measurement_record_length  The size of HEM measurement block.
+ * @param  measurement_hash_algo      Indicates the measurement hash algorithm.
+ *                                    It must align with measurement_hash_alg
+ *                                    (SPDM_ALGORITHMS_MEASUREMENT_HASH_ALGO_*)
+ * @param  measure_exten_log          A pointer to MEL.
+ * @param  mel_size                   The the size of MEL.
+ *
+ * @retval true  the input MEL verify pass.
+ * @retval false the input MEL verify failed.
+ **/
+static bool libspdm_verify_mel(void *measurement_record, uint32_t measurement_record_length,
+                               uint32_t measurement_hash_algo,
+                               void *measure_exten_log, size_t mel_size)
+{
+    spdm_measurement_block_dmtf_t *spdm_hem;
+    size_t hash_size;
+    spdm_measurement_extension_log_dmtf_t *measurement_extension_log;
+    uint8_t verify_hem[LIBSPDM_MAX_MEASUREMENT_EXTENSION_LOG_SIZE + LIBSPDM_MAX_HASH_SIZE];
+    spdm_mel_entry_dmtf_t *mel_entry;
+    uint8_t index;
+
+    spdm_hem = (spdm_measurement_block_dmtf_t *)measurement_record;
+    hash_size = libspdm_get_measurement_hash_size(measurement_hash_algo);
+
+    measurement_extension_log = (spdm_measurement_extension_log_dmtf_t *)measure_exten_log;
+
+    if (mel_size != (sizeof(spdm_measurement_extension_log_dmtf_t) +
+                     measurement_extension_log->mel_entries_len)) {
+        return false;
+    }
+
+    libspdm_zero_mem(verify_hem,
+                     LIBSPDM_MAX_MEASUREMENT_EXTENSION_LOG_SIZE + LIBSPDM_MAX_HASH_SIZE);
+
+    mel_entry = (spdm_mel_entry_dmtf_t *)((uint8_t *)measurement_extension_log +
+                                          sizeof(spdm_measurement_extension_log_dmtf_t));
+
+    for (index = 0; index < measurement_extension_log->number_of_entries; index++) {
+        if (mel_entry->meas_index == spdm_hem->measurement_block_common_header.index) {
+
+            libspdm_copy_mem(
+                verify_hem + hash_size, measurement_extension_log->mel_entries_len,
+                mel_entry,
+                sizeof(spdm_mel_entry_dmtf_t) +
+                mel_entry->measurement_block_dmtf_header.dmtf_spec_measurement_value_size);
+
+            if (!libspdm_measurement_hash_all(
+                    measurement_hash_algo,
+                    verify_hem,
+                    hash_size + sizeof(spdm_mel_entry_dmtf_t) +
+                    mel_entry->measurement_block_dmtf_header.dmtf_spec_measurement_value_size,
+                    verify_hem
+                    )) {
+                return false;
+            }
+        }
+        mel_entry = (spdm_mel_entry_dmtf_t *)
+                    ((uint8_t *)mel_entry + sizeof(spdm_mel_entry_dmtf_t)+
+                     mel_entry->measurement_block_dmtf_header.dmtf_spec_measurement_value_size);
+    }
+
+    if (!libspdm_consttime_is_mem_equal(verify_hem,
```

Comment:
The comparison should be moved to for-loop, to support partial match use case.

The function API should be changed to support partial match as well, to return the matched partial MEL length, if it is not a full MEL.

Reply:
Thanks. I have moved the comparation and returned the mel_size:
```
            *mel_size += sizeof(spdm_mel_entry_dmtf_t) +
                         mel_entry->measurement_block_dmtf_header.dmtf_spec_measurement_value_size;
            if (libspdm_consttime_is_mem_equal(verify_hem,
                                               (uint8_t *)spdm_hem +
                                               sizeof(spdm_measurement_block_dmtf_t),
                                               hash_size)) {
                return true;
            }
        }
```

Reply:
Thanks. Please add unit test.

Reply:
Thanks. I have added the unit_test:
```
        /* Successful response, the MEL size become more bigger when get MEL. MEL number change from 3 to 4*/
        cmocka_unit_test(libspdm_test_requester_get_measurement_extension_log_case3),
        /* Successful response, the MEL size become more bigger when get MEL. MEL number change from 100 to 105*/
        cmocka_unit_test(libspdm_test_requester_get_measurement_extension_log_case4),
        /* Successful response, the MEL size become more bigger when get MEL. MEL number change from 100 to 200*/
        cmocka_unit_test(libspdm_test_requester_get_measurement_extension_log_case5),
```